src/main/java/com/tagtraum/perf/gcviewer/imp/DataReaderSun1_6_0G1.java

```diff
@@ -578,9 +580,12 @@ else if (type.getCollectionType().equals(CollectionType.VM_OPERATION)) {
                     parseDetailEventsIfExist(line, pos, event);
 
                 if (event.getExtendedType().getPattern() == GcPattern.GC_MEMORY_PAUSE) {
-                    setMemoryAndPauses(event, line, pos);
-                }
-                else {
+                    // Support for PrintGCID
```

Comment:
This looks suspicous to me. I admit, that the code is really complicated as it is today - but is this really the right place? Memory and pause parsing should not be different with the PrintGCID option. The gc id is introduced between the startup timestamp and the name of the current event. The pause (and memory details) are after the gc event name:

`2022-08-01T17:17:33.375+0000: 0.163: #0: [GC pause (G1 Evacuation Pause) (young), 0.0087570 secs]`

Reply:
@CuiWeiloong  Please take a look.  Thanks a lot!

Reply:
Done